Update PostgreSQL

qwet1235 opened the pull request:
Provided more examples and edited the existing text to be more clear

darlynnguyen commented on Topics/Tech_Stacks/PostgreSQL_psycopg2.md:
````diff
@@ -45,81 +45,95 @@ A quick and basic runover of PostgreSQL.
+    SELECT sport_name, team_name FROM sports, teams;
 ```
-This operation, first, performs a cross product between table1 and table2, where every row of one table is matched with every other row of the other table. Then it selects the data present in the new cross-products table from columns col1_table1 and col2_table2.
+First, this query performs a cross product between `sports` and `teams`. Each row of table `sports` is matched with each row of the table `teams`. Then, it selects the data from the columns `sport_name` and `team_name` of every row present in the newly joined table.
````
What exactly does match mean? Which attributes are included in the result? I think an example would provide more value to explain this concept over words.

qwet1235 replied:
I craeted new instances for each table and also added more details for the explanations. Can you check if it looks better now?

darlynnguyen commented on Topics/Tech_Stacks/PostgreSQL_psycopg2.md:
````diff
@@ -45,81 +45,95 @@ A quick and basic runover of PostgreSQL.
+SELECT sport_name, team_name FROM sports, teams WHERE sports.sport_id = teams.sport_id;
 ```
-Here it only shows data from the previous table where the value of a row in the column "col1_table1" is greater than 1. This is the syntax for when the table and column names between table 1 and table 2 are unique.
+Here it only shows data from the previous table where the value of `sport_id` from `sports` matches that of `teams`.
````
I think quickly highlighting the usefulness by providing a high level description of what this operation is doing would be helpful here.

qwet1235 replied:
I added more information for this operation. Do you think it is easier to understand now?

darlynnguyen reviewed (changes requested):
Overall, there's better readability and the updated examples and instructions are very helpful!

qwet1235 commented on Topics/Tech_Stacks/PostgreSQL_psycopg2.md:
```diff
@@ -45,81 +45,95 @@ A quick and basic runover of PostgreSQL.
-4. Type "CREATE TABLE table_name (column1 int PRIMARY KEY, ...);", where table_name is the desired name of the table. Inside the round braces is a list of the columns that you want in the table. For each column, you must specify a name and datatype. For example, a table can be created like this:
+1. Execute `psql` to initialize the PostgreSQL environment.
+2. Execute `\c dbname` where `dbname` is the name of the desired database
+3. Run `SET search_path TO schema_name;` where `schema_name` is the name of the schema you want to put your table in.
```
Done!

qwet1235 commented on Topics/Tech_Stacks/PostgreSQL_psycopg2.md:
```diff
@@ -45,81 +45,95 @@ A quick and basic runover of PostgreSQL.
 
 ### Create Tables
-After creating your schema you can create tables/relations to perform operations on. 
+After creating your schema you can create tables/relations to perform operations on. A table contains columns, which describes the attributes of the data being stored, and rows, which are record of intances of the table. Each row contains a set of values that corresponds to the columns of the table.
```
Fixed, thank you

qwet1235 commented on Topics/Tech_Stacks/PostgreSQL_psycopg2.md:
```diff
@@ -45,81 +45,95 @@ A quick and basic runover of PostgreSQL.
 
 ### Table Operations
-Moving onto table operations. To select data from a certain column in a table, you can use a query like:
+In SQL, a query is a statement/command used to retrieve or manipulate data in a relational database. To retrieve data from a certain column in a table, you can use the `SELECT` statement and `FROM` clause:
```
I just added that in the introduction for this section. Thanks!

qwet1235 commented:
Created new instances of the tables to make examples more clear and concrete. Also included output tables for each example. Added information on additional operations and expressions, and updated the parts that were unclear.

darlynnguyen commented on Topics/Tech_Stacks/PostgreSQL_psycopg2.md:
```diff
@@ -45,85 +46,252 @@ A quick and basic runover of PostgreSQL.
+| 2 | soccer | 4 | Manchester City F.C. | 2 |
+| 2 | soccer | 5 | Paris Saint-Germain F.C. | 2 |
+
+Although the query is now more targeted and relevant, the two `sport_id` columns are redundant. To get rid of the redundancy, one solution is to change the columns in the `SELECT` clause so that only one of the `sport_id` columns is retrieved:
```
What is meant by relevant? Maybe you can specify a goal and talk about how it is fulfilled by this join operation. For example, the goal could be to see all the players of each sport.

darlynnguyen commented on Topics/Tech_Stacks/PostgreSQL_psycopg2.md:
````diff
@@ -45,85 +46,252 @@ A quick and basic runover of PostgreSQL.
+```
+SELECT * FROM sports NATURAL JOIN teams;
+```
+`NATURAL JOIN` automatically matches and combines the columns with the same name in the tables being joined. Since `sport_id` is an attribute in both `sports` and `teams`, the resulting set includes all unqiue columns from both tables and excludes duplicate columns. The output of these queries are:
````
Typo: "... includes all unique* columns from both ..."

qwet1235 replied:
Thanks for pointing it out

darlynnguyen reviewed (changes requested):
The concrete tables/examples and logical flow of the article make it really easy to follow along, great work! However there is a minor typo and I've made a small suggestion.

darlynnguyen reviewed (changes requested):
Looks good! 

Maybe consider the suggestion below.

darlynnguyen reviewed (approved):
Looks good, ready to merge. However, I cannot do it since I am not invited to repository.
